scrnaseq.Rmd

```diff
@@ -1974,44 +1999,17 @@ The following table contains the top enriched terms per cluster.
 if (markers_found) {
   cat('#### {.tabset} \n \n')
   
-  # Get top ten up over all databases
-  marker_genesets_up_top_enriched = purrr::map(marker_genesets_up_enriched, function(enrichr_results) {
-    purrr::map_dfr(names(enrichr_results), function(n) {
-      enrichr_result_df = cbind(enrichr_results[[n]], list(
-        Database = factor(rep(n, nrow(enrichr_results[[n]])), levels=names(enrichr_results)),
-        Direction = factor(rep("up", nrow(enrichr_results[[n]])), levels=c("up", "down"))
-      ))
-      return(enrichr_result_df)
-    }) %>% head()
-  })
-  
-  # Get down ten up over all databases
-  marker_genesets_down_top_enriched = purrr::map(marker_genesets_down_enriched, function(enrichr_results) {
-    purrr::map_dfr(names(enrichr_results), function(n) {
-      enrichr_result_df = enrichr_results[[n]]
-      if (nrow(enrichr_result_df) > 0 ) {
-        enrichr_result_df$Database = n
-        enrichr_result_df$Direction = "down"
-      } else {
-        enrichr_result_df = data.frame(enrichr_result_df, Database=as.character(), Direction=as.character())
-      }
-      return(enrichr_result_df)
-    }) %>% head()
-  })
-  
-  # Join
-  marker_genesets_top_enriched = purrr::map(values_to_names(names(marker_genesets_down_top_enriched)), function(n) {
-    return(rbind(marker_genesets_up_top_enriched[[n]], marker_genesets_down_top_enriched[[n]]) %>% 
-      dplyr::arrange(-Odds.Ratio))
-    })
+  # Get top ten up and down over all databases per cluster
+  marker_genesets_top_enriched = marker_genesets_enriched %>% dplyr::group_by(Cluster, Direction) %>%
```

Comment:
Ok, so is the difference that we don't have lists any more and rather a bigger table?

Reply:
yes